devtools/cli/src/main/java/io/quarkus/cli/RegistryAddCommand.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkus.cli;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import io.quarkus.cli.registry.BaseRegistryCommand;
+import io.quarkus.cli.registry.RegistryClientMixin;
+import io.quarkus.registry.config.RegistriesConfig;
+import io.quarkus.registry.config.RegistriesConfigLocator;
+import io.quarkus.registry.config.RegistryConfig;
+import io.quarkus.registry.config.json.JsonRegistriesConfig;
+import io.quarkus.registry.config.json.JsonRegistryConfig;
+import io.quarkus.registry.config.json.RegistriesConfigMapperHelper;
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "add", sortOptions = false, showDefaultValues = true, mixinStandardHelpOptions = false, header = "Add a Quarkus extension registry to the registry client configuration", description = "%n"
+        + "This command will add a Quarkus extension registry to the registry client configuration unless it's already present", headerHeading = "%n", commandListHeading = "%nCommands:%n", synopsisHeading = "%nUsage: ", parameterListHeading = "%n", optionListHeading = "Options:%n")
+public class RegistryAddCommand extends BaseRegistryCommand {
+
+    @CommandLine.Mixin
+    protected RegistryClientMixin registryClient;
+
+    @CommandLine.Parameters(arity = "0..1", paramLabel = "REGISTRY-ID[,REGISTRY-ID]", description = "Registry ID to add to the registry client configuration")
+    String registryIds;
+
+    @CommandLine.Option(paramLabel = "CONFIG", names = { "--config" }, description = "Configuration file")
```

Comment:
Does this configuration parameter belong on the RegistryClientMixin instead? (such that any command involving the registry could be instructed to use that config?)